src/activity/activity_heartbeat_manager.rs

```diff
@@ -0,0 +1,86 @@
+use crate::errors::ActivityHeartbeatError;
+use crate::protos::coresdk::{common, ActivityHeartbeat};
+use crate::ServerGatewayApis;
+use crossbeam::channel::{bounded, unbounded, Receiver, Select, Sender, TryRecvError};
+use dashmap::{DashMap, DashSet};
+use futures::FutureExt;
+use std::convert::TryInto;
+use std::future::Future;
+use std::ops::Div;
+use std::sync::Arc;
+use std::{thread, time};
+use tokio::time::{sleep, Sleep};
+
+pub(crate) struct ActivityHeartbeatManager {
+    /// Core will aggregate activity heartbeats and send them to the server periodically.
+    /// This map contains latest heartbeat details for each activity that reported them.
```

Comment:
Always good to say what the key is in maps (and the set below). We should probably just make a task token type at this point, but that can be another PR

Reply:
Done